dataset/src/test/scala/frameless/functions/AggregateFunctionsTests.scala

```diff
@@ -22,55 +22,82 @@ class AggregateFunctionsTests extends TypedDatasetSuite {
   }
 
   test("sum") {
-    def prop[A: TypedEncoder : Numeric : CatalystSummable](xs: List[A])(
+    abstract class Summer[A, B]{
+      implicit def sum(as: List[A]): B
+    }
+
+    def prop[A: TypedEncoder : Numeric, Out: TypedEncoder : Numeric](xs: List[A])(
       implicit
+      summable: CatalystSummable[A, Out],
+      summer: Summer[A, Out],
       eoa: TypedEncoder[Option[A]],
       ex1: TypedEncoder[X1[A]]
     ): Prop = {
       val dataset = TypedDataset.create(xs.map(X1(_)))
       val A = dataset.col[A]('a)
 
-      val datasetSum = dataset.select(sum(A)).collect().run().toList
+      val datasetSum: List[Out] = dataset.select(sum(A)).collect().run().toList
 
       datasetSum match {
-        case x :: Nil => approximatelyEqual(x, xs.sum)
+        case x :: Nil => approximatelyEqual(summer.sum(xs), x)
         case other => falsified
       }
     }
 
-    check(forAll(prop[BigDecimal] _))
-    check(forAll(prop[Long] _))
-    check(forAll(prop[Double] _))
-
-    // doesn't work yet because resulting type is different
-    // check(forAll(prop[Int] _))
-    // check(forAll(prop[Short]_))
-    // check(forAll(prop[Byte]_))
+    // Replicate Spark's behaviour : Ints and Shorts are cast to Long
+    // https://github.com/apache/spark/blob/master/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Sum.scala#L37
```

Comment:
Use `https://github.com/apache/spark/blob/7eb2ca8/sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/aggregate/Sum.scala#L37` instead (or the L37 might be meaningless in a few commits...)